kong/api/routes/configure.lua

```diff
@@ -0,0 +1,28 @@
+local declarative = require("kong.db.declarative")
+local kong = kong
+
+
+return {
+  ["/configure"] = {
+    POST = function(self, db)
+      if kong.db.strategy ~= "off" then
+        kong.response.exit(400, {
+          message = "this endpoint is only available when Kong is " ..
+                    "configured to not use a database"
+        })
+      end
+
+      local dc = declarative.init(kong.configuration)
+
+      local yaml = self.params.config
+      local entities, err = dc:parse_string(yaml, "config.yml", { yaml = true })
```

Comment:
YAML is a superset of JSON, so we actually accept both here. Lua is not accepted here because of security reasons (loading a Lua file can hang the process; in the CLI this is not a problem).

I'd still like to get the actual filename from the request though to trigger proper JSON errors. I'll mark this a TODO.

Reply:
Iirc Penlight comes with a sandbox, can't we use that and have parity between cli and rest interfaces?

Reply:
I can take a look at it later, but any sandbox for safely loading Lua depends on the `debug` library for hooks, which is notoriously buggy in LuaJIT. For now I'll keep Lua for the CLI only.